Tests/ComposableArchitectureMacrosTests/PresentsMacroTests.swift

```diff
@@ -0,0 +1,163 @@
+            return _destination.wrappedValue
+          }
+          set {
+            if _$isIdentityEqual(newValue, _destination.wrappedValue) == true {
+              _destination.wrappedValue = newValue
+            } else {
+              withMutation(keyPath: \.destination) {
+                _destination.wrappedValue = newValue
+              }
+            }
+          }
+        }
+
+        var $destination: ComposableArchitecture.PresentationState<Destination.State> {
+          get {
+            _destination.projectedValue
+          }
+          set {
+            _destination.projectedValue = newValue
+          }
+        }
+
+        @ObservationStateIgnored private var _destination = ComposableArchitecture.PresentationState<Destination.State>(wrappedValue: nil)
+      }
+      """#
+    }
+  }
+
+  func testPublicAccess() {
+    assertMacro {
+      """
+      public struct State {
+        @Presents public var destination: Destination.State?
+      }
+      """
+    } expansion: {
+      #"""
+      public struct State {
+        public var destination: Destination.State? {
+          @storageRestrictions(initializes: _destination)
+          init(initialValue) {
+            _destination = PresentationState(wrappedValue: initialValue)
+          }
+          get {
+            access(keyPath: \.destination)
+            return _destination.wrappedValue
+          }
+          set {
+            if _$isIdentityEqual(newValue, _destination.wrappedValue) == true {
+              _destination.wrappedValue = newValue
+            } else {
+              withMutation(keyPath: \.destination) {
+                _destination.wrappedValue = newValue
+              }
+            }
+          }
+        }
+
+        public var $destination: ComposableArchitecture.PresentationState<Destination.State> {
+          get {
+            _destination.projectedValue
+          }
+          set {
+            _destination.projectedValue = newValue
+          }
+        }
+
+        @ObservationStateIgnored private var _destination = ComposableArchitecture.PresentationState<Destination.State>(wrappedValue: nil)
+      }
+      """#
+    }
+  }
+
+  func testObservableStateDiagnostic() {
+    assertMacro([
+      ObservableStateMacro.self,
+      ObservationStateIgnoredMacro.self,
+      ObservationStateTrackedMacro.self,
+      PresentsMacro.self,
+    ]) {
+      """
+      @ObservableState
+      struct State: Equatable {
+        @PresentationState var destination: Destination.State?
+      }
+      """
+    } diagnostics: {
+      """
+      @ObservableState
+      struct State: Equatable {
+        @PresentationState var destination: Destination.State?
+        ┬─────────────────
+        ╰─ 🛑 '@PresentationState' property wrapper cannot be used directly in '@ObservableState'
+           ✏️ Use '@Presents' instead
+      }
+      """
+    } fixes: {
+      """
+      @ObservableState
+      struct State: Equatable {
+        @Presents
```

Comment:
The `FixItApplier` we took from the SwiftSyntax repo does not appear to be up to task here. We should check to see if the latest upstream version has any bug fixes would address this.